internal/extension/smartagentextension/config.go

```diff
@@ -58,6 +59,10 @@ func customUnmarshaller(componentViperSection *viper.Viper, intoCfg interface{})
 	}
 	config.Collectd.BundleDir = config.BundleDir
 
+	if config.Collectd.ConfigDir == "/var/run/signalfx-agent/collectd" {
```

Comment:
This prevents using the default value which may be specified by the user config if they are using an existing agent installation (edge case).  You can check for it being set in the `allSettings` map before it's coerced to a saconfig.Config and simply use this "was specified" check instead of tracking the default value.

Reply:
You can also open these changes in another PR unless they are necessary (not sure they are).

Reply:
Will address these changes in another PR.